test/extended/dns/dns.go

```diff
@@ -554,4 +558,59 @@ var _ = Describe("[sig-network-edge] DNS", func() {
 		pod := createDNSPod(f.Namespace.Name, cmd)
 		validateDNSResults(f, pod, expect, times)
 	})
+
+	It("should answer queries using the local DNS endpoint", func() {
+		ctx := context.Background()
+
+		By("starting an exec pod for running queries")
+
+		queryPodExec, err := exutil.NewPodExecutor(oc, "dnsquery", image.ShellImage())
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		queryPod, err := f.ClientSet.CoreV1().Pods(oc.Namespace()).Get(ctx, "dnsquery", metav1.GetOptions{})
+		if err != nil {
+			e2e.Failf("unable to get query pod that was just started: %v", err)
+		}
+		e2e.Logf("DNS query pod is on node: %q" + queryPod.Spec.NodeName)
+
+		By("finding the exec pod's local DNS endpoint")
+
+		// We find the local dns pod by filtering for pods in openshift-dns namespace that are on the same node,
+		// then making sure the pod is running, not being deleted, and is owned by the default dns daemonset.
+		nodeSelector := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector("spec.nodeName", queryPod.Spec.NodeName).String()}
+		dnsPodList, err := f.ClientSet.CoreV1().Pods("openshift-dns").List(ctx, nodeSelector)
+		if err != nil {
+			e2e.Failf("unable to get local DNS pod: %v", err)
+		}
+		localDnsPod := kapiv1.Pod{}
+		for _, pod := range dnsPodList.Items {
+			if pod.Status.Phase == kapiv1.PodRunning && pod.DeletionTimestamp == nil {
+				for _, ownerReference := range pod.ObjectMeta.OwnerReferences {
+					if ownerReference.Name == "dns-default" {
+						localDnsPod = pod
+						break
+					}
+				}
+			}
+		}
+		if reflect.DeepEqual(localDnsPod, kapiv1.Pod{}) {
```

Comment:
Using a Boolean would be simpler.

Reply:
Yea I did this because `localDnsPod` was not a pointer and couldn't figure how to detect non-pointer type as nil. Changing to pointer and detecting nil that way.